### PR TITLE
feat: expose partialStructuredClone

### PR DESCRIPTION
**Pull request by jdalton**
feat: expose `partialStructuredClone`

### CODE REVIEW

**Comment by jdalton** [comment]
This PR is not adding any new logic just exposing a method previously implemented.

File: packages/near-membrane-shared/src/Array.ts [hunk 1]
```diff
@@ -45,6 +44,7 @@ export const {
     find: ArrayProtoFind,
     includes: ArrayProtoIncludes,
     sort: ArrayProtoSort,
+    shift: ArrayProtoShift,
```
Comment (arcadeteddy) [inline packages/near-membrane-shared/src/Array.ts:47]: Above sort.

File: packages/near-membrane-shared/src/ArrayBuffer.ts [hunk 2]
```diff
@@ -0,0 +1,8 @@
+import { ObjectLookupOwnGetter } from './Object';
+
+const ArrayBufferCtor = ArrayBuffer;
```
Comment (arcadeteddy) [inline packages/near-membrane-shared/src/ArrayBuffer.ts:3]: Why is this called Ctor btw. Is it not `ArrayBuffer.constructor`, why can't we just use ArrayBuffer then? I guess the same with Date, etc. Like we don't do it for Boolean, so I'm just confused when we do this.

Comment (jdalton) [reply packages/near-membrane-shared/src/ArrayBuffer.ts:3]: `ArrayBuffer` is `ArrayBuffer.prototype.constructor`

Comment (jdalton) [reply packages/near-membrane-shared/src/ArrayBuffer.ts:3]: We do it any time we store a reference to the Capital Letter constructor and reuse it. See `ArrayCtor`, `ProxyCtor`, etc.

File: packages/near-membrane-shared/src/Array.ts (hunk 1 above)
Comment (jdalton) [reply packages/near-membrane-shared/src/Array.ts:46]: Fixed :)

File: packages/near-membrane-shared/src/ArrayBuffer.ts (hunk 2 above)
Comment (jdalton) [reply packages/near-membrane-shared/src/ArrayBuffer.ts]: Ah I see. When extracted to this repo the reuse no longer applies so I can clean that up. Fixed.

**Review by arcadeteddy — APPROVED** [review]

**Pull request merged**